### PR TITLE
feat(client): ts-migration for the Mobile Layout component

### PR DESCRIPTION
**Pull request by Ismailtlem**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Related to the issue  #43778

### CODE REVIEW

**Review by Sembauke — DISMISSED** [review]

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Can we rename the file to `mobile-layout.tsx`? I believe that's the convention we've been using.

**Review by naomi-lgbt — DISMISSED** [review]
Looks good to me!

File: client/src/templates/Challenges/classic/mobile-layout.tsx [hunk 1]
```diff
@@ -17,27 +16,26 @@ const mapStateToProps = createStructuredSelector({
...
-class MobileLayout extends Component {
+interface MobileLayoutProps {
+  currentTab: number;
+  editor: JSX.Element | null;
+  guideUrl: string;
+  hasPreview?: boolean;
+  instructions?: JSX.Element;
+  moveToTab: typeof moveToTab;
+  preview?: JSX.Element;
+  testOutput?: JSX.Element;
```
Comment (ojeytonwilliams) [inline client/src/templates/Challenges/classic/mobile-layout.tsx:34]:
```suggestion
  hasPreview: boolean;
  instructions: JSX.Element;
  moveToTab: typeof moveToTab;
  preview: JSX.Element;
  testOutput: JSX.Element;
```
Do these need to be optional?

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Hey @Ismailtlem this looks good, I just had one question about optional types:

File: client/src/templates/Challenges/classic/mobile-layout.tsx (hunk 1 above)
Comment (Ismailtlem) [reply client/src/templates/Challenges/classic/mobile-layout.tsx:34]: I don't think so @ojeytonwilliams

**Review by ojeytonwilliams — APPROVED** [review]
LGTM 👍

File: client/src/templates/Challenges/classic/mobile-layout.tsx (hunk 1 above)
Comment (ojeytonwilliams) [reply client/src/templates/Challenges/classic/mobile-layout.tsx]: Thanks @Ismailtlem

**Review by Sembauke — APPROVED** [review]

**Comment by Sembauke** [comment]
Hey @Ismailtlem, thank you for taking this on!

**Pull request merged**